Fred

Fred-18 opened the pull request.

Fred-18 committed:
:wrench: Add new controller

Fred-18 committed:
:recycle: refacto

Fred-18 committed:
:recycle: controller wip

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/AuthorController.java:
```diff
@@ -19,12 +17,23 @@ public class AuthorController {
+    public void saveAuthor(@RequestBody Author author){
+        as.addAuthor(author);
+    }
+    @PostMapping("/author/delete")
```
tu indique path variable mais rien dans la route, ca va planter

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/AuthorController.java:
```diff
@@ -19,12 +17,23 @@ public class AuthorController {
+        as.addAuthor(author);
+    }
+    @PostMapping("/author/delete")
+    public void deleteAuthor(@PathVariable("id") Author author){
```
L'id est de type int

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/AuthorController.java:
```diff
@@ -19,12 +17,23 @@ public class AuthorController {
+    }
     @GetMapping("/author/{firstName}")
     public List <Author> getAuthorByFirstName(@PathVariable("firstname")String firstName){
         return as.getAuthorByFirstName(firstName);
```
Pas de method getAuthorByFirstName dans le service

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/AuthorController.java:
```diff
@@ -19,12 +17,23 @@ public class AuthorController {
         return as.getAuthorByFirstName(firstName);
     }@GetMapping("/author/{lastName}")
     public List <Author> getAuthorByLastName(@PathVariable("lastname")String lastName){
         return as.getAuthorByLastName(lastName);
```
Pas de methode getAuthorByLastName dans le service

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/CategoryController.java:
```diff
@@ -0,0 +1,25 @@
+    @Autowired
+    private CategoryService cs;
+    
+    @GetMapping("/categoty/{id}")
```
typo "categoty"

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/CommentController.java:
```diff
@@ -0,0 +1,31 @@
+    public void saveComment(@RequestBody Comment comment){
+        cos.addComment(comment);
+    }
+    @PostMapping("/comment/delete")
```
pas de path variable dans la route

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/CommentController.java:
```diff
@@ -0,0 +1,31 @@
+        cos.addComment(comment);
+    }
+    @PostMapping("/comment/delete")
+    public void deleteComment(@PathVariable("id") Comment comment){
```
id de type int

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/PublisherController.java:
```diff
@@ -0,0 +1,29 @@
+    public void savePublisher(@RequestBody Publisher publisher){
+        ps.addPublishers(publisher);
+    }
+    @PostMapping("/publisher/delete")
```
pas de path variable

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/PublisherController.java:
```diff
@@ -0,0 +1,29 @@
+        ps.addPublishers(publisher);
+    }
+    @PostMapping("/publisher/delete")
+    public void deletePublisher(@PathVariable("id") Publisher publisher){
```
id de type int

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/PublisherController.java:
```diff
@@ -0,0 +1,29 @@
+    private PublisherService ps;
+    
+    @GetMapping("/publisher/{id}")
+     public Publisher getPublisherById(int id){
```
@PathVarible manquant

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/CommentController.java:
```diff
@@ -0,0 +1,31 @@
+    private CommentService cos;
+    
+    @GetMapping("/comment/{id}")
+    public Comment getCommentById(int id){
```
@PathVarible manquant

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/UserController.java:
```diff
@@ -0,0 +1,28 @@
+    @Autowired
+    UserService us;
+    @GetMapping("/user/{id}")
+    public User getUserById(int id){
```
@PathVarible manquant

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/UserController.java:
```diff
@@ -0,0 +1,28 @@
+    public void saveUser(@RequestBody User user){
+        us.addUser(user);
+    }
+    @PostMapping("user/delete")
```
pas de path variable dans la route

CarbeDev commented on src/main/java/com/labibliotheque/la_bibliotheque/controllers/UserController.java:
```diff
@@ -0,0 +1,28 @@
+        us.addUser(user);
+    }
+    @PostMapping("user/delete")
+    public void deleteUser(@PathVariable("id") User user){
```
Id de type int

Fred-18 committed:
:adhesive_bandage: fix Mr retunr

Fred-18 committed:
:recycle: refacto controler

Fred-18 committed:
:recycle: refacto service

Fred-18 committed:
:bulb: : guillaume comment

Fred-18 committed:
:art: line break

Fred-18 committed:
:white_check_mark: get AuthorById test

Fred-18 committed:
:pencil2: typo post mapping to delete mapping

Fred-18 committed:
:art: Add class Author entity
      Add class Book entity
      Add class Cate entity
      Add class Comment entity
      Add class Comment entity
      Add class Publisher entity
      Add class Role entity
      Add class User entity

Fred-18 committed:
:art: refacto the other class after  introducing entitys class

Fred-18 committed:
:art: Add Mapper

Fred-18 committed:
:recycle: refacto repositorys

Fred-18 committed:
:recycle: refacto service with a new method mapper

Fred-18 committed:
:recycle: refacto repositorys

Fred-18 committed:
:recycle: refacto controllers

Fred-18 committed:
:heavy_plus_sign: add new dependencies

Fred-18 committed:
:lock: config web security

Fred-18 committed:
:sparkles: add securities files

Fred-18 committed:
:seedling add .env file

Fred-18 committed:
:recycle: typo AuthorController

Fred-18 committed:
:recycle: typo BookController

Fred-18 committed:
:sparkles: add CategoryController

Fred-18 committed:
:sparkles: add Category method

Fred-18 committed:
:sparkles: add jwt pas sur de le conserver en raison de firebase voir avec guillaume

Fred-18 committed:
:sparkles: add JwtTokenFilter

Fred-18 committed:
:sparkles: add firebase (Cette version n'est pas foomctionnelle)

Fred-18 committed:
:heavy_plus_sign: add springdoc

Fred-18 committed:
:memo: add reademe (Pas sur de conserver cette version)

Fred-18 committed:
:wrench: ass security config

Fred-18 committed:
:sparkles: add frontend (Ce front n'est pas fonctionnelle

Fred-18 committed:
:pencil2: books with  s

Fred-18 committed:
:sparkles: add swagger update author class

Fred-18 committed:
:seedling: add  update Book model book controller and book mapper

Fred-18 committed:
:seedling: add  update category entity book controller and book mapper

Fred-18 committed:
:seedling: add  update comment controller

Fred-18 committed:
:seedling: add  update PublisherController

Fred-18 committed:
:seedling: add  update UserController

Fred-18 committed:
:white_check_mark: add author test

Fred-18 committed:
:fire: remove files

Fred-18 committed:
:seedling: update bookmapper

Fred-18 committed:
:seedling: remove todo and space

Fred-18 committed:
:sparkles: jwt class

Fred-18 committed:
:art: remove space

Fred-18 committed:
:arrow_up:add dependencices

Fred-18 committed:
:seedling: class role change for be an enum

Fred-18 committed:
:seedling: update user entity,repository,user service

Fred-18 committed:
:seedling: wip book class

Fred-18 committed:
:seedling: auto increment

Fred-18 committed:
:construction: :: docker img

Fred-18 committed:
:seedling: :: modification authros

Fred-18 committed:
:seedling: :: modification Book

Fred-18 committed:
:seedling: :: modification category

Fred-18 committed:
:seedling: :: modification Comment

Fred-18 committed:
:sparkles: ::login form

Fred-18 committed:
:arrow_up: :: modification pomxl

Fred-18 committed:
:seedling: ::modification publisher

Fred-18 committed:
:seedling: ::modification enum

Fred-18 committed:
:sparkles: :: add SimpleBook class

Fred-18 committed:
:seedling: :: modification User

Fred-18 committed:
:seedling: :: modification User

Fred-18 committed:
:wrench: :: add CrossingOrigin

Fred-18 committed:
:sparkles: :: add passwordEncoder

Fred-18 committed:
:sparkles: :: add new signIn methode

Fred-18 committed:
:wrench: add user with relation with book

Fred-18 committed:
:wrench: update jwt

Fred-18 committed:
:wrench: update User

Fred-18 committed:
:wrench: update web security config
